### PR TITLE
Revamp Starlight showcase

### PR DESCRIPTION
**Pull request by delucis**
#### What kind of changes does this PR include?

- Changes or translations of Starlight docs site content

#### Description

- Adds screenshots of sites to our showcase page
- Uses the new `<LinkCard>` component for community plugins
- Preview: https://deploy-preview-547--astro-starlight.netlify.app/showcase/

| Before | After |
|---|---|
| Screen Shot 2023-08-22 at 12 55 36 | Screen Shot 2023-08-22 at 12 55 27 |
| Screen Shot 2023-08-22 at 12 56 27 | Screen Shot 2023-08-22 at 12 56 33 |

### CODE REVIEW

File: docs/src/content/docs/showcase.mdx [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+	<LinkCard
+		href="https://github.com/HiDeoo/starlight-typedoc"
+		title="starlight-typedoc"
+		description="Generate Starlight pages from TypeScript using TypeDoc."
+	/>
+</CardGrid>
+
+## Videos
+
+- Watch Ben and Chris of Astro build [Starwoof](https://starwoof.vercel.app/) with Starlight in real time! [Twitch stream: Astro x Starlight - BLAZING FAST ECO-FRIENDLY NON-GMO ORGANIC docs](https://www.twitch.tv/videos/1841159960)
```
Comment (TheOtterlord) [inline docs/src/content/docs/showcase.mdx:120]: Looks like this VOD isn't available anymore
```suggestion
```

**Review by TheOtterlord — APPROVED** [review]
This looks amazing :partying_face:

File: docs/src/content/docs/showcase.mdx (hunk 1 above)
Comment (delucis) [reply docs/src/content/docs/showcase.mdx:120]: Ohhhh, I had no idea they could expire like that! Another reason to dislike Twitch 😅 

Good catch — thanks @TheOtterlord!

**Review by sarah11918 — APPROVED** [review]
Looks great on both my desktop and mobile!

**Review by HiDeoo — APPROVED** [review]
This is a superb work :star_struck:

**Pull request merged**